### PR TITLE
Log validateResponse errors

### PR DESCRIPTION
**Pull request by pelegdvir**
Log invalid responses (wrong content-type) instead of just raising them (We want to log failures)

### CODE REVIEW

File: src/clients/openapi.js [hunk 1]
```diff
@@ -94,7 +99,17 @@ export function http(req, serviceName, operationName) {
         return axios(
             request,
         ).then((response) => {
-            validateResponse(response);
+            const [validResponse, message] = validateResponse(response);
+            if (!validResponse) {
+                if (logFailure) {
+                    const error = {
```
Comment (pelegdvir) [inline src/clients/openapi.js]: Note: we don't log the response body (unlike failed responses) - we don't want to log maintenance html pages.

File: src/clients/openapi.js [hunk 2]
```diff
@@ -66,11 +66,13 @@ export function buildHeaders(context, req) {
     return headers;
 }
 
-function validateResponse(response) {
+export function validateResponse(response) {
```
Comment (pelegdvir) [inline src/clients/openapi.js]: I've removed the `throw` part from this function to `http` function - to behave in similar way to other logs / exceptions.

File: src/clients/openapi.js [hunk 3]
```diff
@@ -66,11 +66,13 @@ export function buildHeaders(context, req) {
...
     const contentType = get(response, 'headers["content-type"]');
     if (contentType !== 'application/json') {
-        throw new Error(`${contentType} is not a valid response content-type`);
+        const message = `${contentType} is not a valid response content-type`;
```
Comment (omri-globality) [inline src/clients/openapi.js]: Not sure why you need two variables here, looks like the message is enough.
You can send back the error message if there was an error , if there wasn't then return an empty message (null).

Comment (pelegdvir) [reply src/clients/openapi.js]: I'm thinking about the next errors - that are not response type. Maybe it's too much - but...

**Pull request merged**